main.cpp

```diff
@@ -216,33 +267,47 @@ float MidiController::adjustVolume(int midiMessage)
 
 void MidiController::changeBank(int bankNumber)
 {
-    std::cout << "bank" << bankNumber << "selected\n.";
+    std::cout << "Bank " << bankNumber << " selected.\n";
 }
 
 //Thing 5) 
 struct Filter
 {
     std::string filterType = "LP";
-    float frequencyRange = 20000.0F;
+    float maxCutOffFreq = 20000.0F;
     int rezAmount = 1; //1-100
     int modAmount = 1; //1-100
-    float cutOffFrequency = 20.0f;
-    void getSoundTone(float cutoffFrequency = 20.0f,float freqRangeRemaining = 0.0f);
+    float cutOffFrequency;
+    float freqRangeRemaining =0.0f;
+
+    Filter();
+    void getSoundTone(float newCutOffFrequency);
     int addResonance(int rezAmount, int resonance); //returns the total resonance amount
     void cutLowFreq(float cutOffFrequency = 1000.0f);
 };
 
-void Filter::getSoundTone(float cutoffFrequency, float freqRangeRemaining)
+Filter::Filter()
 {
-    if (cutOffFrequency < frequencyRange)
+    filterType = "LP";
+    maxCutOffFreq = 20000.0F;
+    rezAmount = 1; //1-100
+    modAmount = 1; //1-100
+    cutOffFrequency = 0.0f;
+    freqRangeRemaining = 0.0f;
+}
+ 
+void Filter::getSoundTone(float newCutOffFrequency)
+{
+    if (newCutOffFrequency < maxCutOffFreq)
     {
-        freqRangeRemaining = (frequencyRange - cutOffFrequency);
-        std::cout << freqRangeRemaining;
+        freqRangeRemaining = (maxCutOffFreq - newCutOffFrequency);
+        std::cout << freqRangeRemaining << std::endl;
     }
     else
     {
-        std::cout << cutoffFrequency;
+        std::cout << "You are at max cutoff" << std::endl;
     } 
+    newCutOffFrequency = cutOffFrequency; //setting new cut off to cut off.  Will this get erased by the constructor?
```

Comment:
No, because you can't use member functions until AFTER an object has been constructed. An object is considered constructed after the class's constructor has finished executing.